apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/springwebmvc/SpringTransactionNameInstrumentation.java

```diff
@@ -114,19 +121,23 @@ public static void setTransactionName(@Advice.Argument(2) Object handler) {
                 className = handler.getClass().getSimpleName();
                 methodName = null;
             }
-            setName(transaction, className, methodName);
+
+            if (webConfig.isUsePathAsName() || (methodName == null && !className.equals("ResourceHttpRequestHandler"))) {
+                // when method name is not known, using path is a better default as multiple distinct requests
+                // would be handled by a single class. If delegating to a Servlet, then servlet naming will be applied
+                // thus it allows to provide a better fallback in case method name is null and there is no servlet executed.
+                //
+                // ResourceHttpRequestHandler is handling static resources, thus rather high cardinality most of the time
+                // and grouping on a single transaction is preferable.
+                TransactionNameUtils.setNameFromHttpRequestPath(request.getMethod(), request.getServletPath(), request.getPathInfo(), transaction.getAndOverrideName(PRIO_LOW_LEVEL_FRAMEWORK + 1), webConfig.getUrlGroups());
+            } else {
+                TransactionNameUtils.setNameFromClassAndMethod(className, methodName, transaction.getAndOverrideName(PRIO_HIGH_LEVEL_FRAMEWORK));
+            }
```

Comment:
I don't think we need to check `webConfig.isUsePathAsName()`. Instead, the algorithm should be:
```
StringBuilder transactionName;
if (methodName == null) {
  transactionName = getAndOverrideName(PRIO_LOW_LEVEL_FRAMEWORK + 1, false);
} else {
  transactionName = getAndOverrideName(PRIO_HIGH_LEVEL_FRAMEWORK);
}
TransactionNameUtils.setNameFromClassAndMethod(className, methodName, transactionName);
```

By using `getAndOverrideName()` with `false` and the same priority used by `ServletTransactionHelper#applyDefaultTransactionName` to set the path-based name, you separate the logic of setting by path or by class/class-name. The latter WILL override the name because it uses `getAndOverrideName()` with `true`.

Reply:
After testing a bit, we must still check `isUsePathAsName()` and always set the high-level framework name here.
- the naming applied with `ServletTransactionHelper#applyDefaultTransactionName` will only set the low-level name to a path if `isUsePathAsName()` is true
- at spring controller level, the class/method can be either a servlet or another controller, thus we have no guarantee that the default naming will have higher priority.

Of course, we can revisit this later if required.